### PR TITLE
feat(pythonJob): Support HDFS pyFiles dependencies in cluster

### PR DESCRIPTION
**Pull request by gao634209276**
On the cluster side, PythonJob dependencies need to be installed
into the python instance to be used for running the jobs,
or added to the list of paths in config, This change supports
pythonJob HDFS type dependencies use config:
spark.context-settings.launcher.py-files.

**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

---
This change is [Reviewable](https://reviewable.io/reviews/spark-jobserver/spark-jobserver/1180)

### CODE REVIEW

**Comment by bsikander** [comment]
+1 for the change.
test coverage seems to be a bit low but I will review this change soon.

File: job-server-extras/src/main/scala/spark/jobserver/python/PythonContextFactory.scala [hunk 1]
```diff
@@ -134,10 +134,18 @@ trait DefaultContextLikeImplementations {
   override lazy val pythonPath: Seq[String] = pythonPaths(config)
 
   protected def pythonPaths(config: Config): Seq[String] = {
-    val envPaths = sys.env.get("PYTHONPATH").map(_.split(":").toSeq).getOrElse(Seq())
+    // If Spark driver not in cluster, we use SJSPYTHONPATH
+    val pythonPath = if (!sparkContext.deployMode.equals("cluster")) {
+      sys.env.get("SJSPYTHONPATH")
```
Comment (valan4ik) [inline job-server-extras/src/main/scala/spark/jobserver/python/PythonContextFactory.scala]: I am not using Python part, but I don't see any mention of `SJSPYTHONPATH` in documentation or anywhere else before.
Wouldn't it be breaking/unexpected change of the python path?

Comment (valan4ik) [reply job-server-extras/src/main/scala/spark/jobserver/python/PythonContextFactory.scala]: Ah, sorry. I see you define it:
```
      // If the Spark driver is not in cluster, use the PYTHONPATH of client process
      val sjsPythonPath = environment.get("PYTHONPATH", "")
      env.put("SJSPYTHONPATH", sjsPythonPath)
```

Don't pay attention to my comment :)

File: job-server/src/main/scala/spark/jobserver/util/ManagerLauncher.scala [hunk 2]
```diff
@@ -35,76 +85,99 @@ class ManagerLauncher(systemConfig: Config, contextConfig: Config, masterAddress
...
+    }
 
-      for (e <- Try(contextConfig.getConfig("launcher"))) {
-         e.entrySet().asScala.map { c =>
-           // Supervise mode was already handled above, no need to do it here again
-           if (c.getKey != "spark.driver.supervise") {
-            launcher.addSparkArg("--conf", s"${c.getKey}=${c.getValue.unwrapped.toString}")
-           }
-         }
+    getStringSeq(contextConfig, "jars").foreach(launcher.addJar)
```
Comment (bsikander) [inline job-server/src/main/scala/spark/jobserver/util/ManagerLauncher.scala]: Are `jars` and `files` new parameters to context? I do not recall any such parameters previously

**Comment by bsikander** [comment]
@gao634209276 ping

**Comment by bsikander** [comment]
@gao634209276 ping

**Pull request closed without merging**